File: hackerrank/algorithms/diagonalDifference.py
# https://www.hackerrank.com/challenges/diagonal-difference/problem?isFullScreen=true

def diagonalDifference(arr):
    length = len(arr)
    result = 0
    for i in range(0, length):
        result = result + (arr[i][i] - arr[i][length-1-i])
    return abs(result)
        
arr = [[11, 2, 4], [4, 5, 6], [10, 8, -12]]
print(diagonalDifference(arr))

File: hackerrank/diagonalDifference.java
// https://www.hackerrank.com/challenges/diagonal-difference/problem

class Result {

    public static int diagonalDifference(List<List<Integer>> arr) {
        Integer left = 0;
        Integer right = 0;

        for (int i = 0; i < arr.size(); i++) {
            left += arr.get(i).get(i);
            right += arr.get(i).get(arr.size() - 1 - i);
        }   

        return Math.abs(left - right);
    }

}